### DISCUSSION TITLE
Giving users the option to pick from a list of methods in signals

### DISCUSSION THREAD

**Discussion by Braveo** [godotengine/godot-proposals#5500] [Editor]
I love using signals to connect events together, but I found the process itself to be a tad but unintuitive, since you have to type the name of the function you want to connect in order to work. I was wondering if there was a way to either create a new function or connect signal to an existing function via some sort of dropdown list. The best way I can describe it is through this image:

![image](https://user-images.githubusercontent.com/23423003/192397598-1c69de5a-dc03-4b32-82c9-c009469eff47.png)

That way, flicking a lever in a game can connect to an existing door_open method rather than forgetting what the name of the method is and having to go back to the script to remember it. Obviously, I'm probably just being lazy, but I feel like this would speed the process up by a lot. 

Unity has done this in UnityEvents, connecting events with specific methods, and I think this would be a great addition to Godot in Signals. Obviously, these are two different game engines, but even without engine bias in mind, I think this would be a nice addition.

EDIT: Not only that, but I think being able to connect methods such as play(string) in AnimationPlayer quicker could make that even more intuitive. I also wish I could ignore certain call arguments and stuff like that as well, but I'm not sure if that's already possible.

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/5193.
